subsys/bluetooth/mesh/settings.c

```diff
@@ -2424,12 +2442,16 @@ void bt_mesh_clear_cdb_app_key(struct bt_mesh_cdb_app_key *key)
 }
 
 int bt_mesh_model_data_store(struct bt_mesh_model *mod, bool vnd,
-			     const void *data, size_t data_len)
+			     u16_t key, const void *data, size_t data_len)
 {
+	u16_t mod_key;
 	char path[20];
 	int err;
 
-	encode_mod_path(mod, vnd, "data", path, sizeof(path));
+	mod_key = (((u16_t)mod->elem_idx << 8) | mod->mod_idx);
+
+	snprintk(path, sizeof(path), "bt/mesh/%s/%x/%s/%x",
+			vnd ? "v" : "s", mod_key, "data", key);
 
 	if (data_len) {
 		mod->flags |= BT_MESH_MOD_DATA_PRESENT;
```

Comment:
The `MOD_DATA_PRESENT` flag doesn't make sense anymore. It's currently only used to keep track of whether data should be deleted, but this isn't universal anymore, since you might want to delete only one of the model data entries.

I think it can be removed, and lines 2457-2465 can be reduced to `err = settings_save_one(path, data, data_len);`.